rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -554,6 +575,76 @@ class SSAAllocator {
         assert(i);
         return allocation.count(i) == 0;
     }
+
+  private:
+    struct StackArgVisitor {
+        typedef std::function<void(Value*)> ValueAction;
+        static void run(Instruction* i, ValueAction action) {
+            auto runReverseArgs = [&](CallInstructionI* call) {
+                std::vector<Value*> args(call->nCallArgs());
+                call->eachCallArg([&](Value* a) { args.push_back(a); });
+                while (!args.empty()) {
+                    action(args.back());
+                    args.pop_back();
+                }
+            };
+            switch (i->tag) {
+            case Tag::MkArg: {
+                // just the eager value, if it's not missing
+                auto mkarg = MkArg::Cast(i);
+                if (mkarg->hasEagerArg())
```

Comment:
You could instead of `hasEagerArg` and `eagerArg` have the following api:
```
InstrArg::ifEager(std::function<void(Value*)> maybe) {
  if (val() != Missing)
    maybe(val());
}
```
And then you can write:

`MkArg::Cast(i)->arg<0>.ifEager(action);`